src/ring_theory/graded_algebra/homogeneous_ideal.lean

```diff
@@ -0,0 +1,296 @@
+/-
+Copyright (c) 2021 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Eric Wieser
+-/
+
+import ring_theory.ideal.basic
+import ring_theory.ideal.operations
+import linear_algebra.finsupp
+import ring_theory.graded_algebra.basic
+
+/-!
+
+# Homogeneous ideal of a graded algebra
+
+This file defines homogeneous ideals of `graded_algebra A` where `A : ι → submodule R R`and
+operations on them:
+* `mul`, `inf`, `Inf` of homogeneous ideals are homogeneous;
+* `⊤`, `⊥`, i.e. the trivial ring and `R` are homogeneous;
+* `radical` of a homogeneous ideal is homogeneous.
+-/
+
+noncomputable theory
+
+section is_homogeneous_ideal_defs
+
+open set_like direct_sum set
+open_locale big_operators
+
+variables {ι R : Type*} [comm_ring R] [decidable_eq ι] [add_comm_monoid ι]
+variables (A : ι → submodule R R) [graded_algebra A]
+variable (I : ideal R)
+
+/--An `I : ideal R` is called homogeneous if for every `r ∈ I`, every homogeneous component of `r`
+  is in `I`.-/
+def is_homogeneous_ideal : Prop :=
```

Comment:
I suggest calling this `ideal.is_homogeneous`.

Reply:
I have renamed to `ideal.is_homogeneous`